### PR TITLE
ci: Add and document BASE_BUILD_DIR

### LINKED ISSUE #18768: ci: RUN_UNIT_TESTS_SEQUENTIAL is broken on Windows
If the `RUN_UNIT_TESTS_SEQUENTIAL` variable is set to `true` the following message is printed in the Travis log:

```
bash: /home/travis/build/hebasto/bitcoin/build/bitcoin-*/src/test/test_bitcoin: No such file or directory
```

and no tests actually are executed.

Related: #18562

### PR DESCRIPTION
**Pull request by maflcko**
Also fixes #18768

### CODE REVIEW

**Comment by practicalswift** [comment]
Concept ACK

**Comment by hebasto** [comment]
Concept ACK.

**Comment by hebasto** [comment]
Wasn't `BASE_BUILD_DIR` removed in #17549?

**Comment by MarcoFalke** [comment]
Back then it was an alias for BASE_ROOT_DIR (the root of the git folder) and removed because it was redundant.

**Comment by hebasto** [comment]
> Also fixes #18768

Is [this message](https://travis-ci.org/github/hebasto/bitcoin/jobs/679743620#L3321) expected:
```
it looks like wine32 is missing, you should install it.
multiarch needs to be enabled first.  as root, please
execute "dpkg --add-architecture i386 && apt-get update &&
apt-get install wine32"
```

**Comment by MarcoFalke** [comment]
no idea. This was removed in 59e9688eda4c4b01ee1713625632cd766c1a7ca9

Seems unrelated to this pull.

File: ci/test/00_setup_env.sh [hunk 1]
```diff
@@ -49,8 +49,10 @@ export CCACHE_DIR=${CCACHE_DIR:-$BASE_SCRATCH_DIR/.ccache}
 # The depends dir.
 # This folder exists on the ci host and ci guest. Changes are propagated back and forth.
 export DEPENDS_DIR=${DEPENDS_DIR:-$BASE_ROOT_DIR/depends}
-# Folder where the build is done (bin and lib).
+# Folder where the build result is put (bin and lib).
 export BASE_OUTDIR=${BASE_OUTDIR:-$BASE_SCRATCH_DIR/out/$HOST}
+# Folder where the build is done (dist and out-of-tree build).
+export BASE_BUILD_DIR=${BASE_OUTDIR:-$BASE_SCRATCH_DIR/build}
```
Comment (hebasto) [inline ci/test/00_setup_env.sh:55]: Did you mean
```suggestion
export BASE_BUILD_DIR=${BASE_BUILD_DIR:-$BASE_SCRATCH_DIR/build}
```
?

**Comment by hebasto** [comment]
It is absolutely unrelated, but would you mind adding a typo fix in https://github.com/bitcoin/bitcoin/blob/fa6630d11a5a30f73e2e76d0804f518b9cba094d/ci/test/00_setup_env.sh#L62 s/`:`/`:-`/ ?

**Review by hebasto — APPROVED** [review]
ACK 1937d1ea445e9d10408941d3eca95bbca0ad6ac5

**Comment by MarcoFalke** [comment]
Done

**Review by hebasto — APPROVED** [review]
re-ACK fae1a37bc55fc85cd179cdfb35f0b8b92dd69008, only a [typo](https://github.com/bitcoin/bitcoin/pull/18735#issuecomment-619652699) fixed since the [previous review](https://github.com/bitcoin/bitcoin/pull/18735#pullrequestreview-400581536).

**Comment by MarcoFalke** [comment]
The "typo fix" doesn't pass, so I went ahead and reverted it. https://cirrus-ci.com/task/4815052404424704?command=ci#L20

I don't understand bash well enough to debug this, but this might be related: https://wiki.bash-hackers.org/syntax/pe#use_a_default_value

**Comment by hebasto** [comment]
> The "typo fix" doesn't pass, so I went ahead and reverted it. https://cirrus-ci.com/task/4815052404424704?command=ci#L20
> 
> I don't understand bash well enough to debug this, but this might be related: https://wiki.bash-hackers.org/syntax/pe#use_a_default_value

I'll look into it.
Also https://www.gnu.org/software/bash/manual/html_node/Shell-Parameter-Expansion.html

**Review by hebasto — APPROVED** [review]
re-ACK fae49f6e424f31e93c5620d5ff893fb517ef4a8b, which is essentially the same as the previously [reviewed changes](https://github.com/bitcoin/bitcoin/pull/18735#pullrequestreview-400581536).

**Comment by MarcoFalke** [comment]
* https://cirrus-ci.com/build/5921186234236928 passing. Example: https://cirrus-ci.com/task/5984865600405504?command=ci#L2587
* Example travis: https://travis-ci.org/github/bitcoin/bitcoin/jobs/680541926#L3106

**Pull request merged**